### PR TITLE
docs(config): fix configuration file formatting and accuracy

### PR DESCRIPTION
**Pull request by Olawoyin365**
This PR addresses formatting issues, content errors, and structural issues impacting readability in the Configuration File documentation (`ramalama.conf.5.md`).

## Issue Summary

The current configuration documentation page is quite hard to follow and has few breaks, broken links, content errors, manpage-style formatting not adapted for Docusaurus, no clear section organizations, poor scanability etc as identified in issue [#111](https://forge.fedoraproject.org/commops/interns/issues/111) on Fedora Forge

### Before (Current Documentation)

image

### After (This PR)

image

## Video Demonstration of Local Testing of the Web Pages

https://github.com/user-attachments/assets/7a6f4ec7-4342-46ce-8e3d-21dcdbcd2947

## Terminal Formatting

Native Manpage Structural Alignment Standardized the `ramalama.conf.5.md`
 documentation to perfectly align with the core `.1` framework. Stripped out proprietary Docusaurus UI wrappers (e.g., :::note / :::warning) and YAML frontmatter, replacing them with readable Markdown syntax that rendered the man pages good. This guarantees flawless native rendering in both the web portal and the `go-md2man` terminal output for man pages

### Video Demonstration of Local Testing Terminal man pages

https://github.com/user-attachments/assets/abd5f52f-f951-45fb-81da-bd948bc3aa1b

## Summary by Sourcery

Rewrite and restructure the RamaLama configuration file documentation for Docusaurus, improving readability, navigation, and accuracy while adding a complete configuration example.

Documentation:
- Convert the manpage-style ramalama.conf documentation to Docusaurus format with frontmatter, sections, notes, and syntax-highlighted code blocks.
- Clarify configuration options, environment variable overrides, and defaults with organized tables and per-field reference descriptions.
- Add a full example ramalama.conf file to illustrate typical configuration usage.

### CODE REVIEW

**Comment by Olawoyin365** [comment]
> Hey - I've found 2 issues, and left some high level feedback:
> 
> * Several sections that were originally documented as array-of-tables (e.g. `[[ramalama.http_client]]`, `[[ramalama.provider]]`, `[[ramalama.provider.openai]]`, `[[ramalama.benchmarks]]`, `[[ramalama.user]]`) are now presented as single tables (e.g. `[ramalama.http_client]`, `[ramalama.provider.openai]`); please double-check the actual config schema and keep the TOML notation consistent so users don’t adopt an invalid structure.
> * In a few option descriptions the declared type or structure is confusing or inconsistent (e.g. `ramalama.provider.openai` is described as `Type: string` but then used as a table, `port` is a string but is conceptually numeric); consider tightening these to match the real types expected by the parser.
> * There are a couple of small phrasing/format issues that could be cleaned up for clarity and polish (for example, “As mentioned above that the configuration file uses…” under Configuration Format, and the missing space in `**Type:**string` for `port`).
> 
> Prompt for AI Agents
> ```
> Please address the comments from this code review:
> 
> ## Overall Comments
> - Several sections that were originally documented as array-of-tables (e.g. `[[ramalama.http_client]]`, `[[ramalama.provider]]`, `[[ramalama.provider.openai]]`, `[[ramalama.benchmarks]]`, `[[ramalama.user]]`) are now presented as single tables (e.g. `[ramalama.http_client]`, `[ramalama.provider.openai]`); please double-check the actual config schema and keep the TOML notation consistent so users don’t adopt an invalid structure.
> - In a few option descriptions the declared type or structure is confusing or inconsistent (e.g. `ramalama.provider.openai` is described as `Type: string` but then used as a table, `port` is a string but is conceptually numeric); consider tightening these to match the real types expected by the parser.
> - There are a couple of small phrasing/format issues that could be cleaned up for clarity and polish (for example, “As mentioned above that the configuration file uses…” under Configuration Format, and the missing space in `**Type:**string` for `port`).
> 
> ## Individual Comments
> 
> ### Comment 1
> <location path="docs/ramalama.conf.5.md" line_range="18" />
> <code_context>
> +## File Locations
> 
> -For user specific configuration it reads
> +RamaLama searches for configuration files in multiple locations, Files processed later override settings from earlier ones.
> 
> -| Paths                                       | Exception |
> </code_context>
> <issue_to_address>
> **suggestion (typo):** Fix comma splice and capitalization in this sentence about configuration file search order.
> 
> For example: "RamaLama searches for configuration files in multiple locations. Files processed later override settings from earlier ones."
> 
> ```suggestion
> RamaLama searches for configuration files in multiple locations. Files processed later override settings from earlier ones.
> ```
> </issue_to_address>
> 
> ### Comment 2
> <location path="docs/ramalama.conf.5.md" line_range="61" />
> <code_context>
> +
> +## Configuration Format
> +
> +As mentioned above that the configuration file uses the [TOML format](https://toml.io). Every option is nested under its table, with no bare options allowed.
> +
> +**Basic TOML structure:**
> </code_context>
> <issue_to_address>
> **suggestion (typo):** Rephrase the awkward "As mentioned above that" construction.
> 
> Consider: "As mentioned above, the configuration file uses the [TOML format](https://toml.io)." and keep the rest of the sentence unchanged.
> 
> ```suggestion
> As mentioned above, the configuration file uses the [TOML format](https://toml.io). Every option is nested under its table, with no bare options allowed.
> ```
> </issue_to_address>
> ```
> 
> Sourcery is free for open source - if you like our reviews please consider sharing them ✨
> * [X](https://twitter.com/intent/tweet?text=I%20just%20got%20an%20instant%20code%20review%20from%20%40SourceryAI%2C%20and%20it%20was%20brilliant%21%20It%27s%20free%20for%20open%20source%20and%20has%20a%20free%20trial%20for%20private%20code.%20Check%20it%20out%20https%3A//sourcery.ai)
> * [Mastodon](https://mastodon.social/share?text=I%20just%20got%20an%20instant%20code%20review%20from%20%40SourceryAI%2C%20and%20it%20was%20brilliant%21%20It%27s%20free%20for%20open%20source%20and%20has%20a%20free%20trial%20for%20private%20code.%20Check%20it%20out%20https%3A//sourcery.ai)
> * [LinkedIn](https://www.linkedin.com/sharing/share-offsite/?url=https://sourcery.ai)
> * [Facebook](https://www.facebook.com/sharer/sharer.php?u=https://sourcery.ai)
> 
> Help me be more useful! Please click 👍 or 👎 on each comment and I'll use the feedback to improve your reviews.

Thanks for the feedback. I’ve reviewed the points raised:

I’ll verify the actual configuration schema in the codebase to ensure correct use of TOML table vs array-of-table notation before making changes.
I’ll correct type inconsistencies (e.g. port, temp, and provider configuration structure) to match expected runtime behavior.
I’ll fix minor formatting and phrasing issues highlighted.

Appreciate the review, I’ll push updates shortly after validating against the actual implementation.

**Comment by amkr6207** [comment]
@Olawoyin365

Thanks for the work on this PR. I reviewed it and found a few major issues that may cause **CI/test failures**:

1. `docs/ramalama.conf.5.md` no longer matches current repo test expectations in `test/unit/test_config_documentation.py`:
   - parser expects `**field**=...`
   - parser expects section markers like `` `[[ramalama]]` ``

2. `rag_image` vs `rag_images` is inconsistent in the new reference/examples (section naming and example assignment do not match).

3. `"<default store>/benchmarks"` should be escaped as `"\<default store>/benchmarks"` for MDX/docsite safety.

Could you please take a look?

Thanks!


**Update:** I checked the Packit/COPR logs. **CI is currently failing in `%check` on**:
- `test_config_fields_in_manpage`
- `test_consistency_between_conf_and_manpage`
from `test/unit/test_config_documentation.py`.

**Comment by Olawoyin365** [comment]
> @Olawoyin365
> 
> Thanks for the work on this PR. I reviewed it and found a few major issues that may cause **CI/test failures**:
> 
> 1. `docs/ramalama.conf.5.md` no longer matches current repo test expectations in `test/unit/test_config_documentation.py`:
>    
>    * parser expects `**field**=...`
>    * parser expects section markers like `` `[[ramalama]]` ``
> 2. `rag_image` vs `rag_images` is inconsistent in the new reference/examples (section naming and example assignment do not match).
> 3. `"<default store>/benchmarks"` should be escaped as `"\<default store>/benchmarks"` for MDX/docsite safety.
> 
> Could you please take a look?
> 
> Thanks!
> 
> **Update:** I checked the Packit/COPR logs. **CI is currently failing in `%check` on**:
> 
> * `test_config_fields_in_manpage`
> * `test_consistency_between_conf_and_manpage`
>   from `test/unit/test_config_documentation.py`.

Thanks for the detailed review, this is really helpful.

I’m currently going through each of the points you raised. I will revert and push an update once everything is verified. Looking forward to further reviews together

File: docs/ramalama.conf.5.md [hunk 1]
```diff
@@ -1,314 +1,786 @@
...
+store = "$HOME/.local/share/ramalama"
+
+[[ramalama.images]]
+CUDA_VISIBLE_DEVICES = "quay.io/ramalama/cuda"
+```
+
+## Configuration Reference
 
-**api**="none"
+### ramalama Table
```
Comment (Njeri-Kimaru) [inline docs/ramalama.conf.5.md]: @Olawoyin365  kindly check if this is applicable;
 according to the formatting guideline the heading format should be; capitalize only the first word and proper nouns in this case Table doesn't follow that as it is not a proper noun.

File: docs/ramalama.conf.5.md [hunk 2]
```diff
@@ -1,314 +1,786 @@
...
 
-**carimage**="registry.access.redhat.com/ubi10-micro:latest"
+---
+
+#### carimage
+
+**Type:** string  
+**Default:** `"registry.access.redhat.com/ubi10-micro:latest"`
+
+OCI model car image used when building and pushing models with `--type=car`.
```
Comment (Njeri-Kimaru) [inline docs/ramalama.conf.5.md]: @Olawoyin365  kindly check if this is applicable;
car image is a Config.Key and according to our formatting guidelines config.keys should be in inline code.

File: docs/ramalama.conf.5.md [hunk 3]
```diff
@@ -1,314 +1,786 @@
...
-| --------------- | ------- |
-| Podman          | "🦭 > " |
-| Docker          | "🐋 > " |
-| No Engine       | "🦙 > " |
-| No EMOJI support| "> "    |
+#### prefix
 
-**port**="8080"
+**Type:** string  
+**Default:** Based on container engine
```
Comment (Njeri-Kimaru) [inline docs/ramalama.conf.5.md:418]: @Olawoyin365  kindly check if this is applicable;
all the 'defaults' have back ticks except for this line.

File: docs/ramalama.conf.5.md [hunk 4]
```diff
@@ -1,314 +1,786 @@
...
+```toml
+[ramalama]
+transport = "huggingface"
+```
 
-**max_retries**=5
+---
 
-The maximum number of times to retry a failed download
+### ramalama.http_client Table
```
Comment (Njeri-Kimaru) [inline docs/ramalama.conf.5.md]: @Olawoyin365 kindly check if this is applicable;
 according to the formatting guideline the heading format should be; capitalize only the first word and proper nouns in this case Table doesn't follow that as it is not a proper noun.

File: docs/ramalama.conf.5.md [hunk 5]
```diff
@@ -1,314 +1,786 @@
...
+**Example:**
+```toml
+[ramalama.provider.openai]
+api_key = "your-openai-api-key"
+```
+
+---
 
-**storage_folder**="\<default store>/benchmarks"
+### ramalama.benchmarks Table
```
Comment (Njeri-Kimaru) [inline docs/ramalama.conf.5.md]: @Olawoyin365  kindly check if this is applicable;
according to the formatting guideline the heading format should be; capitalize only the first word and proper nouns in this case Table doesn't follow that as it is not a proper noun.

File: docs/ramalama.conf.5.md [hunk 6]
```diff
@@ -1,314 +1,786 @@
...
+```toml
+[ramalama.http_client]
+max_retry_delay = 60
+```
 
-**max_retry_delay**=30
+---
 
-The maximum delay between retry attempts in seconds
+### ramalama.provider Table
```
Comment (Njeri-Kimaru) [inline docs/ramalama.conf.5.md]: @Olawoyin365  kindly check if this is applicable;
according to the formatting guideline the heading format should be; capitalize only the first word and proper nouns in this case Table doesn't follow that as it is not a proper noun.

File: docs/ramalama.conf.5.md [hunk 7]
```diff
@@ -1,314 +1,786 @@
...
+**Example:**
+```toml
+[ramalama.benchmarks]
+storage_folder = "/custom/benchmark/results"
+```
+
+---
 
-`[[ramalama.user]]`
+### ramalama.user Table
```
Comment (Njeri-Kimaru) [inline docs/ramalama.conf.5.md]: @Olawoyin365  kindly check if this is applicable;
according to the formatting guideline the heading format should be; capitalize only the first word and proper nouns in this case Table doesn't follow that as it is not a proper noun.

**Comment by Olawoyin365** [comment]
> > @Olawoyin365
> > Thanks for the work on this PR. I reviewed it and found a few major issues that may cause **CI/test failures**:
> > 
> > 1. `docs/ramalama.conf.5.md` no longer matches current repo test expectations in `test/unit/test_config_documentation.py`:
> >    
> >    * parser expects `**field**=...`
> >    * parser expects section markers like `` `[[ramalama]]` ``
> > 2. `rag_image` vs `rag_images` is inconsistent in the new reference/examples (section naming and example assignment do not match).
> > 3. `"<default store>/benchmarks"` should be escaped as `"\<default store>/benchmarks"` for MDX/docsite safety.
> > 
> > Could you please take a look?
> > Thanks!
> > **Update:** I checked the Packit/COPR logs. **CI is currently failing in `%check` on**:
> > 
> > * `test_config_fields_in_manpage`
> > * `test_consistency_between_conf_and_manpage`
> >   from `test/unit/test_config_documentation.py`.
> 
> Thanks for the detailed review, this is really helpful.
> 
> I’m currently going through each of the points you raised. I will revert and push an update once everything is verified. Looking forward to further reviews together

Hi @amkr6207 

Thank you so much for the detailed review and for pointing out those CI failures. I have gone ahead and pushed fixes to address all of those issues while still maintaining the new, readable Docusaurus layout.

Here is a breakdown of the fixes I included in the latest push:

Fixed Parser Expectations (**field**=...): I have injected the **field**="..." format directly underneath the new #### field_name headings. This ensures the regex in 

test_config_documentation.py
 successfully captures all the configuration options at the start of the line.
Added Section Markers: Inserted the backticked `[[section]]` tags right underneath each table heading, ensuring the parser flawlessly understands when it enters a new configuration structure (e.g., `[[ramalama.images]]`).

MDX Escape Fix: Properly escaped <default store> as \<default store> to prevent any Docusaurus/MDX rendering errors.
Fixed Python Test Script Omission: While debugging, I found that the 

test_config_documentation.py
 test script itself was dropping nested fields under the provider and benchmarks blocks, falsely classifying them as globally undocumented arguments. I submitted a tiny fix to subsections_with_fields in the test script so 

test_no_undocumented_fields_in_manpage
 passes successfully.
I pulled the repository locally, ran pytest test/unit/test_config_documentation.py, and can confirm that all 5 checks (including the previously failing 

test_config_fields_in_manpage
 and 

test_consistency_between_conf_and_manpage
) are now passing perfectly.

Please check again from your end to confirm these fixes and let me know if there's anything else required.

**Comment by Olawoyin365** [comment]
Hi @Njeri-Kimaru 

Thanks for taking the time to highlight all of those casing differences 

When formatting the documentation pages, I've split the headers into two distinct categories based on the formatting guidelines (I have updated it in our HackMD too):

Conceptual English Headings: For standard topics like ## Environment Variables or ## Configuration Format, I use Title Case (capitalizing the first letter) because they act like standard book chapters or article sections.

Code Literal Headings: For headers that directly reference code or configuration keys (like ### ramalama table or ### RAMALAMA_CONFIG), I preserve the exact programmatic casing (lowercase for TOML tables, ALL-CAPS for bash environment variables). 

This ensures users don't accidentally copy-paste a capitalized literal like [Ramalama], which would fail because TOML is case-sensitive! (And for the word "table", since it's just a generic noun, it stays lowercase to follow the sentence-case rule you mentioned perfectly).

Thanks again for the sharp eyes on this, it's really helping to improve the documentation

**Comment by Njeri-Kimaru** [comment]
> Hi @Njeri-Kimaru
> 
> Thanks for taking the time to highlight all of those casing differences
> 
> When formatting the documentation pages, I've split the headers into two distinct categories based on the formatting guidelines (I have updated it in our HackMD too):
> 
> Conceptual English Headings: For standard topics like ## Environment Variables or ## Configuration Format, I use Title Case (capitalizing the first letter) because they act like standard book chapters or article sections.
> 
> Code Literal Headings: For headers that directly reference code or configuration keys (like ### ramalama table or ### RAMALAMA_CONFIG), I preserve the exact programmatic casing (lowercase for TOML tables, ALL-CAPS for bash environment variables).
> 
> This ensures users don't accidentally copy-paste a capitalized literal like [Ramalama], which would fail because TOML is case-sensitive! (And for the word "table", since it's just a generic noun, it stays lowercase to follow the sentence-case rule you mentioned perfectly).
> 
> Thanks again for the sharp eyes on this, it's really helping to improve the documentation

You are most welcome @Olawoyin365 and thank you too.

**Comment by amkr6207** [comment]
@Olawoyin365

Thanks for the updates. I re-checked the latest changes.

I still see one major concern:
- `test/unit/test_config_documentation.py` was modified as part of this docs-formatting PR. For issue https://forge.fedoraproject.org/commops/interns/issues/111 , this feels out-of-scope and changes existing validation behavior.

Could you please revert the test-file changes and keep the fix limited to `docs/ramalama.conf.5.md`?

@mikebonnet, @rhatdan, @bmahabirbu could you please confirm whether changing `test/unit/test_config_documentation.py` is appropriate for this issue?

**Comment by amkr6207** [comment]
> @Olawoyin365
> 
> Thanks for the updates. I re-checked the latest changes.
> 
> I still see one major concern:
> 
> * `test/unit/test_config_documentation.py` was modified as part of this docs-formatting PR. For issue https://forge.fedoraproject.org/commops/interns/issues/111 , this feels out-of-scope and changes existing validation behavior.
> 
> Could you please revert the test-file changes and keep the fix limited to `docs/ramalama.conf.5.md`?
> 
> @mikebonnet, @rhatdan, @bmahabirbu could you please confirm whether changing `test/unit/test_config_documentation.py` is appropriate for this issue?

@mikebonnet, @rhatdan, @bmahabirbu could you please confirm whether changing `test/unit/test_config_documentation.py` is appropriate for this issue?

File: docs/ramalama.conf.5.md [hunk 8]
```diff
@@ -736,10 +736,10 @@ The `ramalama.provider` table configures hosted API providers that RamaLama can
 
 #### openai
 
-**openai**=""
+**openai**
```
Comment (praxyfarhana) [inline docs/ramalama.conf.5.md:696]: Hi @Olawoyin365 kindly check if this is applicable, should it be "openAI"

Comment (Olawoyin365) [reply docs/ramalama.conf.5.md:696]: Hi @praxyfarhana, thanks for the observation. I've used openai (lowercase) specifically where it refers to the TOML configuration key (e.g., [ramalama.provider.openai]), as these keys are consistently lowercase in our configuration schema. Using the exact technical casing ensures users don't face parsing issues in their config files.

However, I've used 'OpenAI' in the descriptive prose where I'm referring to the organization. I believe this distinction helps maintain both technical accuracy and professional grammar.

File: docs/ramalama.conf.5.md [hunk 9]
```diff
@@ -430,7 +430,7 @@ prefix = "AI> "
 
 #### port
 
-**Type:** string
+**Type:** integer
 **Default:** `"8080"`
```
Comment (praxyfarhana) [inline docs/ramalama.conf.5.md:443]: Hi @Olawoyin365 have noticed some inconsistencies with ports' data type (Default: `"8080"`).

File: docs/ramalama.conf.5.md [hunk 10]
```diff
@@ -589,7 +589,7 @@ summarize_after = 10
 
 #### temp
 
-**Type:** string  
+**Type:** float 
 **Default:** `"0.8"`
```
Comment (praxyfarhana) [inline docs/ramalama.conf.5.md:609]: Hi @Olawoyin365 have noticed some inconsistencies with data type (Default: "0.8").

File: docs/ramalama.conf.5.md [hunk 11]
```diff
@@ -671,14 +671,14 @@ The `ramalama.provider` table configures hosted API providers that RamaLama can
 
 #### openai
 
-**Type:** string  
+**Type:** table  
 **Default:** `""`
```
Comment (praxyfarhana) [inline docs/ramalama.conf.5.md]: Hi @Olawoyin365 have noticed some inconsistencies with  data type (Default: "").

**Comment by Olawoyin365** [comment]
> > @Olawoyin365
> > Thanks for the updates. I re-checked the latest changes.
> > I still see one major concern:
> > 
> > * `test/unit/test_config_documentation.py` was modified as part of this docs-formatting PR. For issue https://forge.fedoraproject.org/commops/interns/issues/111 , this feels out-of-scope and changes existing validation behavior.
> > 
> > Could you please revert the test-file changes and keep the fix limited to `docs/ramalama.conf.5.md`?
> > @mikebonnet, @rhatdan, @bmahabirbu could you please confirm whether changing `test/unit/test_config_documentation.py` is appropriate for this issue?
> 
> @mikebonnet, @rhatdan, @bmahabirbu could you please confirm whether changing `test/unit/test_config_documentation.py` is appropriate for this issue?

Thanks for the feedback, @amkr6207.

The changes to `test/unit/test_config_documentation.py` were made because this test acts as a documentation linter. Specifically, I had to update the `subsections_with_fields` list and adjust the regex logic to handle the new `[ramalama.provider.openai]` nested structure added. When I updated ramalama.conf.5.md to improve its accuracy, the existing test regexes failed because they didn't account for the new structure and improved key descriptions. Updating the validation suite alongside the documentation is a practice to ensure CI passes and that the docs remain in sync with the codebase. Without these changes, the automated tests would flag several 'undocumented fields' or formatting errors that we have already addressed.

As you have also suggested, I'm happy to wait for the maintainers (@mikebonnet, @rhatdan, @bmahabirbu) to confirm if changes to the `test/unit/test_config_documentation.py` is an accepted practice.

**Comment by Olawoyin365** [comment]
Hi @praxyfarhana, You're right that `port` and `temp` represent numeric values. However, in the RamaLama implementation (
ramalama/config.py), these fields are currently defined as strings in the BaseConfig class. That is, The reason they are strings is that RamaLama stores them as raw text from the configuration source and only converts them to numbers when they are actually used.

The documentation in this PR is intentionally aligned with the codebase to ensure users understand how the configuration is initially parsed and stored from TOML/Environment sources. I've corrected the temp type designation in the manpage to string to be consistent with the other fields and the implementation. Thank you for the detailed review

**Review by Ritika-Bhambri — COMMENTED** [review]
Hi @Olawoyin365, great progress on the formatting! The documentation will look much cleaner with these changes.

While looking at how these changes interact with the CI pipeline, I noticed an edge case in the test suite that we should probably future-proof against regarding the `provider` and `benchmarks` sections.

**The issue:**  
The parser in `test/unit/test_config_documentation.py` appears to rely on regex-based extraction of documented fields (e.g., matching **field_name**). It effectively assumes that any bolded field corresponds to a top-level [ramalama] config unless it is explicitly excluded via the `subsections_with_fields` handling.

Currently, `provider` and `benchmarks` seem to be missing from that exclusion logic in `get_documented_fields_in_manpage()`. Because of this, if a nested field is documented using the **fieldname** syntax, the test may incorrectly treat it as a top-level global field.

**Why this does not fail CI currently:**  
The **api_key** formatting passes CI because api_key exists at the top level of BaseConfig. Since the test compares sets of documented vs. actual config fields, this doesn’t trigger a failure. However, if a provider-specific field like **model** or **base_url** were added, this could cause the test to fail.

Additionally, it seems the parser only matches one level of sections (e.g., [[ramalama.<section>]]) and does not fully track deeper nested sections like [[ramalama.provider.openai]]. Because of this, once the section context is lost, nested fields such as **api_key** may end up being interpreted as top-level fields during parsing.

**Inconsistency:**  
It looks like `benchmarks` is already handled in `get_documented_fields_in_conf()`, but not in `get_documented_fields_in_manpage()`, which introduces an inconsistency between the two parsers.

**Suggestion:**  
To avoid future CI issues as documentation expands, we could:

Option A: Use backticks (e.g., `api_key`) for nested fields so they are ignored by the parser  
Option B: Update the test suite to include `provider` and `benchmarks` in the subsection exclusion logic (and potentially improve handling of nested sections)

It is not a blocker for this PR, but maybe worth addressing to make the documentation pipeline more robust.

Great work on the documentation improvements!

**Comment by rhatdan** [comment]
Please verify tat the man pages actually generate look good.

man ./*.1

**Comment by Olawoyin365** [comment]
> Please verify tat the man pages actually generate look good.
> 
> man ./*.1

@rhatdan,  Thanks for pointing that out. I regenerated and verified the build locally, and the man pages compile successfully and display correctly in the terminal.

Here is a quick recording of the final terminal output.

https://github.com/user-attachments/assets/9267e343-5802-4a6e-b890-794bf86031d8
